Makefile

```diff
@@ -3,38 +3,61 @@ export GO15VENDOREXPERIMENT=1
 SHORT_NAME := deis-e2e
 
 SRC_PATH := /go/src/github.com/deis/workflow-e2e
-DEV_IMG := quay.io/deis/go-dev:0.9.1
-
-RUN_CMD := docker run --rm -e DEIS_ROUTER_SERVICE_HOST=${DEIS_ROUTER_SERVICE_HOST} -e DEIS_ROUTER_SERVICE_PORT=${DEIS_ROUTER_SERVICE_PORT} -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${DEV_IMG}
-DEV_CMD := docker run --rm -e GO15VENDOREXPERIMENT=1 -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${DEV_IMG}
-
-TEST_OPTS := -slowSpecThreshold=120.00 -noisyPendings=false
-PARALLEL_TEST_OPTS := ${TEST_OPTS} -p
 
 MUTABLE_VERSION ?= canary
 VERSION ?= git-$(shell git rev-parse --short HEAD)
 
+GINKGO_NODES ?= 15
+
+TEST_OPTS := -slowSpecThreshold=120.00 -noisyPendings=false
+PARALLEL_TEST_OPTS := ${TEST_OPTS} -nodes=${GINKGO_NODES}
+
 DEIS_REGISTRY ?= quay.io/
 IMAGE_PREFIX ?= deis
 IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
 MUTABLE_IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
 
+JUNIT ?= true
+
+DEV_IMG := quay.io/deis/go-dev:0.9.0
+DEV_CMD_ARGS := --rm -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${DEV_IMG}
+DEV_CMD := docker run ${DEV_CMD_ARGS}
+DEV_CMD_INT := docker run -it ${DEV_CMD_ARGS}
+RUN_CMD := docker run --rm -e DEIS_CONTROLLER_URL=${DEIS_CONTROLLER_URL} \
+													 -e DEFAULT_EVENTUALLY_TIMEOUT=${DEFAULT_EVENTUALLY_TIMEOUT} \
+													 -e MAX_EVENTUALLY_TIMEOUT=${MAX_EVENTUALLY_TIMEOUT} \
+													 -e JUNIT=${JUNIT} \
+													 -e DEBUG=${DEBUG} \
+													 -v ${HOME}/.kube:/root/.kube \
+													 -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${IMAGE}
+
+.PHONY: check-controller-url
+check-controller-url:
+	@if [ -z "$$DEIS_CONTROLLER_URL" ]; then \
+	  echo "DEIS_CONTROLLER_URL is not exported.  You must export this variable to proceed."; \
+	exit 2; \
+	fi
+
+.PHONY: dev-env
+dev-env:
+	${DEV_CMD_INT} bash
+
 .PHONY: bootstrap
 bootstrap:
 	${DEV_CMD} glide install
 
 .PHONY: test-integration
-test-integration:
-	DEFAULT_EVENTUALLY_TIMEOUT="30s" ginkgo ${TEST_OPTS} tests/
-
-.PHONY: test-integration
-test-integration-parallel:
+test-integration: check-controller-url
 	ginkgo ${PARALLEL_TEST_OPTS} tests/
 
+.PHONY: test-integration-serial
+test-integration-serial: check-controller-url
+	ginkgo ${TEST_OPTS} tests/
+
 .PHONY: docker-build
 docker-build:
 	docker build -t ${IMAGE} ${CURDIR}
-	docker tag -f ${IMAGE} ${MUTABLE_IMAGE}
+	docker tag ${IMAGE} ${MUTABLE_IMAGE}
```

Comment:
it'll be deprecated in future releases, but if you're using an older version of Docker (like 1.9.1), this will fail. Let's keep this in for now and we can come back to it in the future.